pkg/views/gitprovider/select.go

```diff
@@ -125,6 +128,60 @@ func GitProviderSelectionView(gitProviderAddView *apiclient.SetGitProviderConfig
 	if err != nil {
 		log.Fatal(err)
 	}
+
+	commitsigningDataForm := huh.NewForm(huh.NewGroup(
```

Comment:
What's the reason for this being a separate form?  Couldn't these just be groups inside the `userDataForm`?

Reply:
one of the main reason behind this that this will improve the speed of forms. at first I tried adding it to the form but after the form was filled it was taking a lot of time to ```git provider registered ```.

Reply:
@bryans-go the speed of submitting the form is not influenced by how many groups it has. After the user finishes the form, the data is sent to the server who then makes a check on the git provider. Myb you had a connection hiccup that caused this, but it was definitely not influenced by adding these 2 groups to the form above.

Reply:
I forgot to mention that another problem is updating the ```fmt.Print("\033[H\033[2J")
		views.RenderInfoMessage(getGitProviderSigningHelpMessage(gitProviderAddView.Id))```  helper message beacuse the core functionality uses fmt.println .  and also there is no way of updating the message mid-way in the form

Reply:
Ah right. Then I agree. All good.